### ISSUE TITLE
MailArchiveHelper serialise and deserialise do the same thing?

### ISSUE THREAD

**Issue by csperkins** [glasgow-ipl/ietfdata#92]
In `mailhelper_*.py` files the `serialise()` and `deserialise()` functions seem to do the same thing. Should they?

**Comment by lumisota** [comment]
This is true when the value of the metadata field is the same when serialised and deserialised. For example, all of the fields provided by ```mailhelper_headerdata.py``` are strings, and serialised to strings. 

Where they are different, ```serialise()``` builds a dictionary mapping field names to string representations of the metadata. For example, in ```mailhelper_datatracker.py```, the fields are ietfdata.Datatracker types: ```serialise()``` builds a map to their resource URIs, and ```deserialise()``` uses these to reconstruct the objects.

It might be cleaner/more efficient for all metadata fields to be JSON serialisable, but that might give less flexibility later on.

**Comment by lumisota** [comment]
That said, will push a fix for the method signatures. They should be:

```
    serialise(self, metadata: Dict[str, Any]) -> Dict[str, str]
  deserialise(self, metadata: Dict[str, str]) -> Dict[str, Any]
```

**Comment by lumisota** [comment]
> That said, will push a fix for the method signatures. They should be:
> 
> ```
>     serialise(self, metadata: Dict[str, Any]) -> Dict[str, str]
>   deserialise(self, metadata: Dict[str, str]) -> Dict[str, Any]
> ```

This isn't right -- can serialise to/from any JSON type. I've submitted a PR that uses "Any" in lieu of a JSON type alias (an actual JSON type alias would be recursively defined, and that isn't straightforward).

**Comment by lumisota** [comment]
Fixed by #93.
